### PR TITLE
LPS-190136 Implements ObjectValidationRuleEngine interface and Create an Object Validation Client Extension Sample

### PR DESCRIPTION
**Pull request by mateussgn**
Related issues:
https://liferay.atlassian.net/browse/LPS-190136
https://liferay.atlassian.net/browse/LPS-190137

**How to generate the liferay-sample-etc-spring-boot.zip:**
A local plugin needs to be generated because we will need to use it to build this client extension:
1. In **/liferay-portal/modules/sdk/gradle-plugins-workspace**
        and execute 
        `gradlew deploy`.

2. In **/liferay-portal/modules/sdk/gradle-plugins-workspace/build/tmp/jar**
    will be the plugin **com.liferay.gradle.plugins.workspace-x.x.x.jar**
    you will need to get its paths to use in the next step.

3. Change the [settings.gradle](https://github.com/liferay-objects/liferay-portal/blob/bc3721aa6aefdbe2752ae85b9751ec1fdd0a129b/workspaces/liferay-sample-workspace/settings.gradle#L3-L4) file with the path you copied in the last step.
    3.1. Change the **...enire path...** with the appropriate path.

**Deploying all client extensions inside liferay-sample-etc-spring-boot:**
**Since this is a scoped validation sample you must follow these steps strictly**
1. In **/liferay-portal/workspaces/liferay-sample-workspace** execute:
    `gradlew :client-extensions:liferay-sample-etc-spring-boot:deploy`

3. The zip will be generated in **/liferay-portal/workspaces/liferay-sample-workspace/bundles/osgi/client-extensions**

**Instructions for testing:**

1. Paste the zip file to **/bundles/osgi/client-extensions**.
    1.1. If the client-extension folder does not exist, just start the Portal and the directory should be created.
    1.2. Once the file is in the **/bundles/osgi/client-extensions** directory just wait and it will work automatically.

2. Go to **/liferay-portal/workspaces/liferay-sample-workspace**
    2.1. Run the command:
            `gradlew :client-extensions:liferay-sample-etc-spring-boot:bootRun`
    2.2. It will start the client extension sample.
    2.3. Perform this request in your browser: **http://localhost:58081/ready**. If you see a page that says "READY" it worked.

4. Create a new Custom Object Definition named "Employee".

5. Add a new Custom Object Field named "name" from the Text type.

6. Add a new Object Validation named "Employee Validation" form type "function#liferay-sample-etc-spring-boot-object-validation-1".

7. Active the new Object Validation.

8. Publish the new Employee Custom Object Definition.

9. Create a new Object Entry for the new Employee Custom Object Definition.
    8.1. Fill the name field with the string "Invalid Name".
        8.1.1. The Object Entry must not be created.
    8.2. Fill the name field with any other string.
        8.2.1. The Object Entry must be created.

### CODE REVIEW

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Review by pedro-oliveira446 — APPROVED** [review]

**Comment by MarinhoFeliphe** [comment]
Hey @mateussgn see my comments [here](https://github.com/mateussgn/liferay-portal/pull/104#pullrequestreview-1559814931)

Thanks!

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java [hunk 1]
```diff
@@ -235,14 +236,15 @@ public void unassociateObjectField(ObjectField objectField) {
...
+			long companyId, long objectValidationRuleId, boolean active,
+			String engine, Map<Locale, String> errorLabelMap,
+			Map<Locale, String> nameMap, String outputType, String script,
 			List<ObjectValidationRuleSetting> objectValidationRuleSettings)
 		throws PortalException {
 
 		_validate(
-			engine, nameMap, outputType, script, objectValidationRuleSettings);
+			companyId, engine, nameMap, outputType, script,
+			objectValidationRuleSettings);
```
Comment (MarinhoFeliphe) [inline modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:247]: @mateussgn I believe we can get the companyId from [objectValidationRule](https://github.com/liferay-objects/liferay-portal/blob/ec0bdc7086b4a10dbfd33420c112540a2981678b/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java#L249), this way we can avoid to change a lot of places, and it seems safe to me, what do you think?

Thanks!

**Review by MarinhoFeliphe — CHANGES_REQUESTED** [review]

File: modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java (hunk 1 above)
Comment (mateussgn) [reply modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java:247]: I agree with you! I'll make the necessary changes.

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:sf

**Review by MarinhoFeliphe — APPROVED** [review]
LGTM!! :tada:

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:object

**Comment by victorhcunha** [comment]
ci:test:object

**Comment by victorhcunha** [comment]
PR reject for the same reason as #2367

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:reevaluate:1387401_8560

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:test:object

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:stop

**Comment by victorhcunha** [comment]
ci:test:object

**Comment by victorhcunha** [comment]
ci:test:relevant

**Comment by victorhcunha** [comment]
ci:test:object

**Comment by mateussgn** [comment]
ci:stop

**Comment by mateussgn** [comment]
ci:test:sf

**Comment by mateussgn** [comment]
ci:test:relevant

**Comment by mateussgn** [comment]
After https://github.com/brianchandotcom/liferay-portal/pull/139312 be merged, I'll forward this PR.

**Comment by mateussgn** [comment]
ci:close

**Comment by mateussgn** [comment]
Resent: https://github.com/liferay-objects/liferay-portal/pull/2471

**Comment by guilhermedcamacho** [comment]
See #2475

**Pull request closed without merging**